storm-core/src/jvm/org/apache/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -231,6 +231,18 @@ public void deleteBlob(String key, Subject who) throws AuthorizationException, K
     }
 
     @Override
+    public void clearRemovedBlob(String key) throws AuthorizationException {
+        validateKey(key);
+        // not pulling recent change and checking ACL
```

Comment:
if you handle this in deleteBlob itself, we don't need to add new api to blobstore. i.e. if getStoredBlobMeta throws KeyNotFoundException, just log it and skip the ACL check and proceed with deleting the keys.